source/Sample/Program.cs

```diff
@@ -1,50 +1,92 @@
-                // send an event to tell chrome to send us all Page events
-                // but we only subscribe to certain events in this session
-                var pageEnableResult = chromeSession.SendAsync<ChromeDevTools.Protocol.Chrome.Page.EnableCommand>().Result;
-                Console.WriteLine("PageEnable: " + pageEnableResult.Id);
-                chromeSession.Subscribe<Protocol.Chrome.Page.DomContentEventFiredEvent>(domContentEvent =>
+                        Width = ViewPortWidth,
+                        Height = ViewPortHeight
+                    });
+
+                    var navigateResponse = await chromeSession.SendAsync(new NavigateCommand
                     {
-                        Console.WriteLine("DomContentEvent: " + domContentEvent.Timestamp);
+                        Url = "http://www.google.com"
                     });
-                // you might never see this, but that's what an event is ... right?
-                chromeSession.Subscribe<Protocol.Chrome.Page.FrameStartedLoadingEvent>(frameStartedLoadingEvent =>
+                    Console.WriteLine("NavigateResponse: " + navigateResponse.Id);
+
+                    // STEP 4 - Register for events (in this case, "Page" domain events)
+                    // send an command to tell chrome to send us all Page events
+                    // but we only subscribe to certain events in this session
+                    var pageEnableResult = await chromeSession.SendAsync<Protocol.Chrome.Page.EnableCommand>();
+                    Console.WriteLine("PageEnable: " + pageEnableResult.Id);
+
+                    chromeSession.Subscribe<LoadEventFiredEvent>(loadEventFired =>
                     {
-                        Console.WriteLine("FrameStartedLoading: " + frameStartedLoadingEvent.FrameId);
+                        // we cannot block in event handler, hence the task
+                        Task.Run(async () =>
+                        {
+                            Console.WriteLine("LoadEventFiredEvent: " + loadEventFired.Timestamp);
+
+                            var documentNodeId = (await chromeSession.SendAsync(new GetDocumentCommand())).Result.Root.NodeId;
+                            var bodyNodeId =
+                                (await chromeSession.SendAsync(new QuerySelectorCommand
+                                {
+                                    NodeId = documentNodeId,
+                                    Selector = "body"
+                                })).Result.NodeId;
+                            var height = (await chromeSession.SendAsync(new GetBoxModelCommand {NodeId = bodyNodeId})).Result.Model.Height;
+
+                            await chromeSession.SendAsync(new SetVisibleSizeCommand {Width = ViewPortWidth, Height = height});
+
+                            Console.WriteLine("Taking screenshot");
+                            var screenshot = await chromeSession.SendAsync(new CaptureScreenshotCommand {Format = "png"});
```

Comment:
I do like this change ... it does seem like more advanced example.  Would you be opposed to making a Simple and Advanced sample?  Then we can link them from the README.

Reply:
Do you mean keeping the original sample too?

Reply:
Yeah ... maybe we can even make it simpler

Reply:
Something for later though